core/org.osate.xtext.aadl2.ui/src/org/osate/xtext/aadl2/ui/outline/Aadl2OutlineTreeProvider.java

```diff
@@ -168,8 +176,19 @@ protected boolean isLeaf(final EObject modelElement) {
 			return modelElement.eContainer() instanceof RecordValue;
 		} else if (modelElement instanceof IntegerLiteral) {
 			return false;
-		} else {
-			return super.isLeaf(modelElement);
+		}
+		else {
+			return !Iterables.any(modelElement.eClass().getEAllContainments(), containmentRef -> {
+				if (containmentRef.getEReferenceType() == Aadl2Package.eINSTANCE.getRealization()
```

Comment:
I would introduce a variable for `containmentRef.getEReferenceType()` to make this easier to read.